internal/oci/runtime_oci.go

```diff
@@ -635,17 +635,46 @@ func waitContainerStop(ctx context.Context, c *Container, timeout time.Duration,
 		if err := kill(pid); err != nil {
 			return fmt.Errorf("failed to kill process: %v", err)
 		}
+	case newTimeout := <-c.stopTimeoutChan:
+		// if a new timeout comes in,
+		// interrupt the old one, and start a new one
+		close(chControl)
+		return waitContainerStop(ctx, c, newTimeout, ignoreKill)
```

Comment:
I don't like recursion in here -- there might be a possibility to abuse it, leading to stack overflow. Maybe better to use `goto`.

Reply:
ugh I really don't love using goto, but a for loop feels excessive for one case. I resign to your suggestion :)